src/main/java/uk/gov/pay/products/resources/ProductResource.java

```diff
@@ -62,4 +63,15 @@ public Response findProduct(@PathParam("productExternalId") String productExtern
                 .orElseGet(() ->
                         Response.status(NOT_FOUND).build());
     }
+
+    @DELETE
```

Comment:
Is there a reason why we consider this a delete, rather than disable. Not that it matters too much, but I would have said keeping the concept of disabling/enabling from resource down would be clearer, and would allow us to eg. introduce an 'enable' endpoint in future without confusing ourselves